src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/LoginHandler.cs

```diff
@@ -0,0 +1,636 @@
+                    await writer.WriteShortAsync(offset, isAsync, ct).ConfigureAwait(false); // password offset
+                    await writer.WriteShortAsync(encryptedPasswordLengthInBytes / 2, isAsync, ct).ConfigureAwait(false);
+                    offset += encryptedPasswordLengthInBytes;
+                }
+                else
+                {
+                    // case where user/password data is not used, send over zeros
+                    await writer.WriteShortAsync(0, isAsync, ct).ConfigureAwait(false);  // userName offset
+                    await writer.WriteShortAsync(0, isAsync, ct).ConfigureAwait(false);
+                    await writer.WriteShortAsync(0, isAsync, ct).ConfigureAwait(false);  // password offset
+                    await writer.WriteShortAsync(0, isAsync, ct).ConfigureAwait(false);
+                }
+
+                await writer.WriteShortAsync(offset, isAsync, ct).ConfigureAwait(false); // app name offset
+                await writer.WriteShortAsync(context.ApplicationName.Length, isAsync, ct).ConfigureAwait(false);
+                offset += context.ApplicationName.Length * 2;
+
+                await writer.WriteShortAsync(offset, isAsync, ct).ConfigureAwait(false); // server name offset
+                await writer.WriteShortAsync(context.ServerName.Length, isAsync, ct).ConfigureAwait(false);
+                offset += context.ServerName.Length * 2;
+
+                await writer.WriteShortAsync(offset, isAsync, ct).ConfigureAwait(false);
+                if (context.UseFeatureExt)
+                {
+                    await writer.WriteShortAsync(4, isAsync, ct).ConfigureAwait(false); // length of ibFeatgureExtLong (which is a DWORD)
+                    offset += 4;
+                }
+                else
+                {
+                    await writer.WriteShortAsync(0, isAsync, ct).ConfigureAwait(false); // unused (was remote password ?)
+                }
+
+                await writer.WriteShortAsync(offset, isAsync, ct).ConfigureAwait(false); // client interface name offset
+                await writer.WriteShortAsync(clientInterfaceName.Length, isAsync, ct).ConfigureAwait(false);
+                offset += clientInterfaceName.Length * 2;
+
+                await writer.WriteShortAsync(offset, isAsync, ct).ConfigureAwait(false); // language name offset
+                await writer.WriteShortAsync(context.Language.Length, isAsync, ct).ConfigureAwait(false);
+                offset += context.Language.Length * 2;
+
+                await writer.WriteShortAsync(offset, isAsync, ct).ConfigureAwait(false); // database name offset
+                await writer.WriteShortAsync(context.Database.Length, isAsync, ct).ConfigureAwait(false);
+                offset += context.Database.Length * 2;
+
+                if (null == s_nicAddress)
+                    s_nicAddress = TdsParserStaticMethods.GetNetworkPhysicalAddressForTdsLoginOnly();
```

Comment:
Nit: I'd personally want to keep this method dedicated to only explicitly showing the network IO, so I'd reference a NicAddress property similar to:
```csharp
public static byte[] NicAddress
    => s_nicAddress ??= TdsParserStaticMethods.GetNetworkPhysicalAddressForTdsLoginOnly();
```